src/pages/QuestionAnswerViewer.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useContext, useEffect } from 'react';
+import {
+  Route, useRouteMatch, useHistory, useParams,
+} from 'react-router-dom';
+
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Alert from '@material-ui/lab/Alert';
+
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Accordion from '@material-ui/core/Accordion';
+import AccordionSummary from '@material-ui/core/AccordionSummary';
+import AccordionDetails from '@material-ui/core/AccordionDetails';
+
+import UserContext from '@/user';
+import API from '@/API';
+import { formatDateTimeNicely } from '@/utils/cache';
+
+import Loading from '@/components/loading/Loading';
+import StoredAnswersetViewer from '@/components/shared/answersetView/StoredAnswersetViewer';
+
+import EditAnswer from '@/pages/EditAnswer';
+
+export default function QuestionAnswerViewer() {
+  const [question, updateQuestion] = useState(null);
+  const [answers, updateAnswers] = useState([]);
+
+  const user = useContext(UserContext);
+
+  const { question_id } = useParams();
+  const { path } = useRouteMatch();
+  const history = useHistory();
+
+  let answer_id;
+  // If we are rendering an answer, get answer_id with useRouteMatch
+  const match = useRouteMatch(`${path}/answer/:answer_id`);
+  if (match) {
+    answer_id = match.params.answer_id;
+  }
+
+  async function fetchQuestion() {
+    let token;
+    if (user) {
+      token = user.id_token;
+    }
+    const response = await API.getQuestion(question_id, token);
+    if (response.status === 'error') {
+      return;
+    }
+    updateQuestion(response);
+  }
+
+  useEffect(() => { fetchQuestion(); }, [user, question_id]);
+
+  async function fetchAnswers() {
+    let token;
+    if (user) {
+      token = user.id_token;
+    }
+    const response = await API.getAnswersByQuestion(question_id, token);
+    if (response.status === 'error') {
+      return;
+    }
+    const newAnswers = response;
+    updateAnswers(newAnswers);
+
+    if (!answer_id && newAnswers.length > 0) {
+      // Set default answer to first
+      history.push(`/question/${question_id}/answer/${newAnswers[0].id}`);
+    }
+  }
+
+  function getAnswer(id) {
+    return answers.find((a) => a.id === id);
+  }
+
+  useEffect(() => { fetchAnswers(); }, [user, question_id, answer_id]);
```

Comment:
This will effectively keep a user from going 'back'. A user goes to '<root>/question/<q_id>' from the question list page and that question has answers, then we are automatically redirected to /question/<q_id>/answer/<a_id>. I want to go back to the question list, so I press back. I go to /question/<q_id> and it redirects me right back to the answer page. I can't go back anymore.